EGaDS! Open Project Game/Assets/Scripts/AbilitySystem/PlayerController.cs

```diff
@@ -104,6 +113,33 @@ private bool IsSideOnWall(Vector2 side)
     private float _wallJumpTimer = -1;
     private float _currentAirJumpCount = 0;
 
+    /// <summary>
+    /// Returns a Vector2 based on the direction of the keys pressed. For dash purposes
+    /// </summary>
+    private Vector2 CheckKeyDirection()
+    {
+        // Checks which movement inputs are being pressed (indicates direction for dash)
+        var xKeys = Input.GetAxisRaw("Horizontal");
+        var yKeys = Input.GetAxisRaw("Vertical");
+        return new Vector2(xKeys, yKeys).normalized;
+    }
+    /// <summary>
+    /// Dashes in a specified direction for a certain period of time (values adjustable by dash parameters in editor)
+    /// </summary>
+    private IEnumerator Dash()
+    {
+        _canDash = false;
+        _isDashing = true;
```

Comment:
Structurally, instead of using boolean triggers, why not create a 'state machine' type thing here, where you track the state of the character with an enum, and have the state be dashing when dash is being ran.